data_structures/src/Stack/Stack.cs

```diff
@@ -6,44 +6,71 @@ namespace ADT
 {
     public class Stack<T> : IStack<T>
     {
+        private T[] _stack { get; set; }
+        private int _lastIndex { get; set; }
+
         public Stack(int size)
         {
-
+            _stack = new T[size]; 
         }
 
         public IEnumerator<T> GetEnumerator()
         {
-            throw new NotImplementedException();
+             return (IEnumerator<T>)_stack.GetEnumerator(); 
         }
 
         public bool IsEmpty()
         {
-            throw new NotImplementedException();
+            return this.Size() > 0 ? false : true;
         }
 
         public T Peek()
         {
-            throw new NotImplementedException();
+            if(!this.IsEmpty())
+            {
+                return _stack[_lastIndex -1];
+            }
+            else
+            {
+                throw new Exception("Stack is empty- Cannot peek.");
+            }
         }
 
         public T Pop()
         {
-            throw new NotImplementedException();
+            if(!this.IsEmpty())
+            {
+                var returnItem =  _stack[_lastIndex -1];
+                _lastIndex--;
+                return returnItem;
+            }
+            else
+            {
+                throw new Exception("Stack is null- cannot pop.");
+            }
         }
 
         public void Push(T item)
         {
-            throw new NotImplementedException();
+            if(_stack.Length > _lastIndex)
+            {
+                _stack[_lastIndex] = item;
+                _lastIndex++;
```

Comment:
You can shorten this line by doing `_stack[_lastIndex++] = item;` Also, if you only have one statement following an if else block, no need for braces, they are noisy.